charts/argo-workflows/templates/server/server-ingress.yaml

```
@@ -0,0 +1,39 @@
+{{- if .Values.server.enabled -}}
+{{- if .Values.server.ingress.enabled -}}
+{{- $serviceName := include "argo-workflows.server.fullname" . -}}
+{{- $servicePort := .Values.server.servicePort -}}
+apiVersion: {{ include "argo-workflows.ingress.apiVersion" . }}
+metadata:
+  name: {{ template "argo-workflows.server.fullname" . }}
+  labels:
+    {{- include "argo-workflows.labels" $ | nindent 4 }}
+    {{- range $key, $value := .Values.server.ingress.labels }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+  annotations:
+    {{- range $key, $value := .Values.server.ingress.annotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+spec:
+  rules:
+    {{- range .Values.server.ingress.hosts }}
+    - host: {{ . }}
+      http:
+        paths:
+          {{- if $.Values.server.ingress.paths }}
+          {{- range $.Values.server.ingress.paths }}
+          - backend:
+              serviceName: {{ .serviceName }}
+              servicePort: {{ .servicePort }}
+          {{- end }}
+          {{- end }}
+          - backend:
+              serviceName: {{ $serviceName }}
+              servicePort: {{ $servicePort }}
```

```suggestion
    {{- $kubeVersion := .Capabilities.KubeVersion.GitVersion}}
    {{- range .Values.server.ingress.hosts }}
    - host: {{ . }}
      http:
        paths:
          {{- if $.Values.server.ingress.paths }}
          {{- range $.Values.server.ingress.paths }}
          {{- include "argo-workflows.ingress.service" (dict "kubeVersion" $kubeVersion "serviceName" .serviceName "servicePort" .servicePort) | nindent 10 }}
          {{- end }}
          {{- end }}
          {{- include "argo-workflows.ingress.service" (dict "kubeVersion" $kubeVersion "serviceName" $serviceName "servicePort" $servicePort) | nindent 10 }}
```

Also see new helper function. The new ingress api changed the fields layout. Since the api field in here was attempting to be backward compatible. I suggested a function to correctly build the backend field.